### PR TITLE
Bugfix for #803 - Debug mode

### LINKED ISSUE #803: debug mode not debug
### Duplicates

- [X] I have searched the existing issues

### Steps to reproduce 🕹

attribute in chat.py is not named correctly

### Current behavior 😯

wrong name can't call the attrb on the object

### Expected behavior 🤔

to use the attrb in a  call without error

### Your prompt 📝

```yaml
# Paste your prompt here

### PR DESCRIPTION
**Pull request by chozzz**
### Background
- Fixes issue #803, and its duplicates;
    - #818
    - #835
    - #858
    - #860
    - #869
    - #887

### Changes
- Fix #803
    - Brings back debug mode
    - Replaces all callees from `cfg.debug` to `cfg.debug_mode` that was updated on 5b2d6010dc59bab1026d13bfcd75b37618e573b9
    - Remove unnecessary config instance at main.py

### Test Plan

I added a test placeholder for the unit test of local cache. I am wondering if we can have the git workflow hook to run basic unit test? @Torantulino 

### PR Quality Checklist
- [x] ~My pull request is atomic and focuses on a single change.~
- [x] I have thouroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [ ] ~I have documented my changes clearly and comprehensively.~
- [x] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

**Comment by mcc0nkie** [comment]
These changes are definitely needed. Thanks, man.

**Comment by KixPanganiban** [comment]
+1 to this! Was thinking of sending in a similar PR, but you already did it first and better. Thanks @chozzz

File: scripts/memory/local.py [hunk 1]
```diff
@@ -28,10 +28,20 @@ class LocalCache(MemoryProviderSingleton):
     def __init__(self, cfg) -> None:
         self.filename = f"{cfg.memory_index}.json"
         if os.path.exists(self.filename):
-            with open(self.filename, 'rb') as f:
-                loaded = orjson.loads(f.read())
-                self.data = CacheContent(**loaded)
+            try:
```
Comment (nponeccop) [inline scripts/memory/local.py]: This one isn't related to the debug flag issue. Submit as a separate PR.

File: tests/local_cache_test.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import unittest
```
Comment (nponeccop) [inline tests/local_cache_test.py]: This one doesn't belong here either.

**Review by nponeccop — CHANGES_REQUESTED** [review]

File: scripts/memory/local.py (hunk 1 above)
Comment (chozzz) [reply scripts/memory/local.py]: Hi @nponeccop I moved this out to https://github.com/Torantulino/Auto-GPT/pull/913

File: tests/local_cache_test.py (hunk 2 above)
Comment (chozzz) [reply tests/local_cache_test.py]: Removed

File: scripts/main.py [hunk 3]
```diff
@@ -292,11 +294,13 @@ def parse_arguments():
...
         cfg.set_smart_llm_model(cfg.fast_llm_model)
 
+    if args.debug:
+        print_to_console("Debug Mode: ", Fore.GREEN, "ENABLED")
+        cfg.set_debug_mode(True)
 
 
 # TODO: fill in llm values here
 check_openai_api_key()
-cfg = Config()
```
Comment (nponeccop) [inline scripts/main.py]: This one is unrelated, submit it apart

File: scripts/main.py [hunk 4]
```diff
@@ -17,6 +17,7 @@
 import argparse
 import logging
 
+# Initialize config
```
Comment (nponeccop) [inline scripts/main.py]: This one is unrelated. Submit it apart.

**Review by nponeccop — CHANGES_REQUESTED** [review]

File: scripts/main.py (hunk 3 above)
Comment (chozzz) [reply scripts/main.py]: retracted

File: scripts/main.py (hunk 4 above)
Comment (chozzz) [reply scripts/main.py]: retracted

**Review by nponeccop — APPROVED** [review]

**Comment by nponeccop** [comment]
@Torantulino Can we fix the master?

**Review by richbeales — APPROVED** [review]
